sdk/core/core/inc/az_str_stream.h

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
```

Comment:
I'm not convinced that this is necessary. Can't it all be done with whatever slice structures/functions we already have?
By the way, we need a function to get the length of a slice (just return its size field).

Reply:
https://github.com/Azure/azure-sdk-for-c/pull/36